pkg/deployments/tasks/20230213-yearn-linear-pool/test/contracts/ReadOnlyReentrancyAttackerYearnLP.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-linear/ILinearPool.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/ERC20Helpers.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+
+/**
+ * @notice Performs a read-only reentrancy attack on a target linear pool, making use of the `receive` callback
+ * in the middle of a Vault operation.
+ */
+contract ReadOnlyReentrancyAttackerYearnLP {
+    using FixedPoint for uint256;
+
+    enum AttackType { SET_TARGETS, SET_SWAP_FEE }
+
+    uint8 public constant RECOVERY_MODE_EXIT_KIND = 255;
+
+    uint256 private constant _LOWER_TARGET = 20e18;
+    uint256 private constant _UPPER_TARGET = 500e18;
+    uint256 private constant _SWAP_FEE_PERCENTAGE = 1e16;
+
+    IVault private immutable _vault;
+    AttackType private _attackType;
+    ILinearPool private _pool;
+
+    constructor(IVault vault) {
+        _vault = vault;
+    }
+
+    /**
+     * @dev Starts attack on target pool. Since LinearPool attacks are permissioned and regular joins and exits are
+     * disabled, the attacker contract must be the owner (or be granted permissions to perform the calls).
+     *
+     * It is possible to enter the Vault with a RecoveryMode exit - but `_handleRemainingEth` is only called if the
+     * Vault is requested to unwrap WETH, which requires a target pool that contains WETH.
+     * For the AAVE specialization the token pair should then be WETH and waWETH, but waETH doesn't seem to comply with
```

Comment:
Same as in #2262 / https://github.com/balancer-labs/balancer-v2-monorepo/pull/2262#discussion_r1107548592:
if the contract is exactly the same, feel free to just delete this source code and use the one in the AAVE fork test for this task, adding a comment where you deploy the contract clarifying that the source is the same.

If this pool accepts WETH, then the attacker test for ERC4626 linear pool would be more suitable.

Finally, if the contract has any difference that I might be missing, just adjust these docs.